platform/kube/inject/inject.go

```diff
@@ -460,6 +463,29 @@ func intoObject(c *Config, in interface{}) (interface{}, error) {
 		}
 	}
 
+	// skip customized namespaces
+	for _, ignoredNamespace := range c.IgnoredNamespaces {
+		if ignoredNamespace == "" {
+			return out, nil
+		}
+
+		if obj.GetNamespace() == ignoredNamespace {
+			// If the namesepace was defined in both IgnoredNamespaces and Namespaces,
+			// then IgnoredNamespaces will overwrite the Namespaces.
+			if c.Namespaces[0] == "" {
+				glog.V(2).Infof("The namespace %s will be ignored by sidecar initializer", ignoredNamespace)
+			} else {
+				for _, namespace := range c.Namespaces {
+					if ignoredNamespace == namespace {
+						glog.V(2).Infof("The namespace %s will be ignored by sidecar initializer", ignoredNamespace)
+						break
```

Comment:
This is unnecessary and will cause confusion. Its better to have a mutually exclusive model.
User can specify either IncludeNamespaces or ExcludeNamespaces, but not both, where the semantics are
if ExcludeNamespaces is provided, sidecars will be injected on ALL except the namespaces provided..
if IncludeNamespaces is provided, sidecars will be injected on ONLY the namespaces provided.

Reply:
Done